File: daily_problems/2026/04/0423/personal_submission/cf106494d_yawn_sean.py
# Submission link: https://codeforces.com/gym/106494/submission/372105581
def main():
    n = II()
    nums = LII()
    nums.sort()
    
    total = 2 * sum(nums)
    ans = total
    
    for i in range(n):
        total -= 2 * nums[i]
        ans = fmax(ans, total + nums[i // 2] + nums[i - i // 2])
    
    print(ans)

File: daily_problems/2026/04/0423/personal_submission/cf106494d_yawn_sean.cpp
#include <bits/stdc++.h>
// #pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/all"

using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

	int n;
	cin >> n;

	vector<int> nums(n);
	long long total = 0;
	for (auto &x: nums) cin >> x, total += x * 2;

	sort(nums.begin(), nums.end());

	long long ans = total;

	for (int i = 0; i < n; i ++) {
		total -= nums[i] * 2;
		ans = max(ans, total + nums[i / 2] + nums[i - i / 2]);
	}

	cout << ans;

	return 0;
}